src/services/control.js

```diff
@@ -219,30 +220,51 @@ var ControlService = {
 
 module.exports = ControlService;
 
-var maybeParseJSON = function (filename, def, callback) {
-  fs.readFile(filename, {encoding: 'utf-8'}, function (err, body) {
-    if (err) {
-      if (err.code === 'ENOENT') {
-        return callback(null, def);
+var readAndMergeConfigFiles = function (files, def, callback) {
+  // for compatibility. Might be called with files as a single path, not an
+  // array of paths
+  if (!Array.isArray(files)) {
+    files = [files];
+  }
+
+  async.reduce(files, {}, function (previousValue, currentValue, reduceCallback) {
+    fs.readFile(currentValue, {encoding: 'utf-8'}, function (err, body) {
+      if (err) {
+        if (err.code === 'ENOENT') {
+          return callback(null, def);
+        }
+
+        return callback(err);
       }
 
-      return callback(err);
-    }
+      var doc;
+      try {
+        doc = JSON.parse(body);
+      } catch (e) {
+        doc = {};
+        logger.warn('Configuration file contained invalid JSON', {
+          errMessage: e.message,
+          filename: currentValue,
+          source: body
+        });
+      }
 
-    var doc;
-    try {
-      doc = JSON.parse(body);
-    } catch (e) {
-      logger.warn('Configuration file contained invalid JSON', {
-        errMessage: e.message,
-        filename: filename,
-        source: body
-      });
+      // I'm surprised this little concatenation loop works as well as it does.
+      // Could definitely use some testing to be sure it covers all the
+      // _reasonable_ use cases.
+      for (var site in doc) {
+        if (doc.hasOwnProperty(site)) {
+          if (previousValue.hasOwnProperty(site)) {
+            previousValue[site] = previousValue[site].concat(doc[site]);
+          } else {
+            previousValue[site] = doc[site];
+          }
+        }
+      }
```

Comment:
This is reasonable, but it works because we have pretty consistent JSON schema across each of the configuration files. It might be a little dangerous if we start messing with that.

Maybe we should use a more general-purpose object merging utility from npm?